astropy/io/fits/convenience.py

```diff
@@ -709,6 +710,98 @@ def info(filename, output=None, **kwargs):
     return ret
 
 
+def diff(inputa, inputb, *args, **kwargs):
+    """
+    Compare two parts of a fits file, including entire FITS files, and
+    FITS HDU objects.
+
+    Parameters
+    ----------
+    inputa : str of `HDUList`
+        The filename of a FITS file on disk, or an `HDULIST` object.
+
+    inputb : str or `HDUList`
+        The filename of a FITS file on disk, or an `HDULIST` object.
+
+    ext
+        The rest of the arguments are for extension specification if your
+        inputA and inputB are string filenames (will not work if inputA and
+        inputB are HDU objects. They are flexible and are best illustrated
+        by examples.
+
+        By extension number::
+            diff('inA.fits', 'inB.fits', 0)      # the primary HDU
+            diff('inA.fits', 'inB.fits', 2)      # the second extension
+            diff('inA.fits', 'inB.fits', ext=2)  # the second extension
+
+        By name, i.e., ``EXTNAME`` value (if unique)::
+            diff('inA.fits', 'inB.fits', 'sci')
+            diff('inA.fits', 'inB.fits', extname='sci')  # equivalent
+
+        Note ``EXTNAME`` values are not case sensitive
```

Comment:
What the heck, I couldn't see some of these these comments before... missed some